File: code/3._BubbleSort/Python/programming.py/Bubble_Sort.py
def bubble_sort(arr):
    """
    Sorts an array using the Bubble Sort algorithm.

    :param arr: The input array to be sorted.
    """
    n = len(arr)
    for i in range(n):
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]


# Example usage:
my_list = [64, 34, 25, 12, 22, 11, 90]
bubble_sort(my_list)
print(my_list)

File: code/3._BubbleSort/JavaScript/programming.js/Bubble_Sort.js
function bubbleSort(arr) {
  /**
   * Sorts an array using the Bubble Sort algorithm.
   *
   * @param arr The input array to be sorted.
   */
  const n = arr.length;
  for (let i = 0; i < n - 1; i++) {
    for (let j = 0; j < n - i - 1; j++) {
      if (arr[j] > arr[j + 1]) {
        // Swap arr[j] and arr[j+1]
        [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
      }
    }
  }
}

const myArray = [64, 34, 25, 12, 22, 11, 90];
bubbleSort(myArray);
console.log(myArray);
